openedx/adg/lms/utils/date_utils.py

```diff
@@ -22,26 +22,32 @@ def month_choices(default_title=None):
     return choices
 
 
-def year_choices(from_year=MINIMUM_YEAR_OPTION, default_title=None):
+def year_choices(from_year=MINIMUM_YEAR_OPTION, default_title=None, is_upcoming=False, to_year=None):
     """
-    Create choices for all the years from `from_year` to current year, in descending order. If `from_year` is not
-    provided then range will start from 1900 i.e. [(2020, 2020) ... (1900, 1900)]
+    Create choices for previous or upcoming years depending on the is_upcoming flag.
 
     Args:
         default_title (str): The default title i.e. Year
-        from_year (int): Optional, range from current year to this year; defaults to 1900
+        is_upcoming (bool): If True, generate list of previous years; else list of upcoming years
+        to_year (int): Optional, range from current year to this year; defaults to 10 year from current year
+        from_year (int): Optional, descending range from this year to current year; defaults to 1900
 
     Returns:
         list: Choice in a list of tuples
     """
     current_year = MAXIMUM_YEAR_OPTION
-
-    if from_year > current_year:
-        raise ValueError('Invalid from year {from_year}, it must be less than {current_year}'.format(
-            from_year=from_year, current_year=current_year)
-        )
-
-    choices = [(year, year) for year in range(current_year, (from_year - 1), -1)]
+    if not to_year:
+        to_year = MAXIMUM_YEAR_OPTION + 10
+
+    if not is_upcoming and from_year > current_year:
+        raise ValueError(f'Invalid from year {from_year}, it must be less than {current_year}')
+    if is_upcoming and to_year < current_year:
+        raise ValueError(f'Invalid to year {to_year}, it must be greater than {current_year}')
+
+    if is_upcoming:
+        choices = [(year, year) for year in range(current_year, to_year + 1)]
+    else:
+        choices = [(year, year) for year in range(current_year, (from_year - 1), -1)]
```

Comment:
If clauses are getting a little confusing. Maybe we can do something like:

```suggestion
if is_upcoming:
	if to_year < current_year:
		raise ValueError()
	choices = [(year, year) for year in range(current_year, to_year + 1)]
else:
	if from_year > current_year:
		raise ValueError()
	choices = [(year, year) for year in range(current_year, (from_year - 1), -1)]
```